### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by Oberacda**
This PR fixes various different bugs.

* Fix missing feedback with action nodes
* Remove catch all finally clause to ensure crashes are not masked
* Fix race condition when setting tree state.

### CODE REVIEW

File: ros_bt_py/ros_bt_py/ros_nodes/action.py [hunk 1]
```diff
@@ -498,7 +498,10 @@ class Action(Leaf):
     _goal_type: type
     _feedback_type: type
     _result_type: type
-    _ac: Optional[Client] = None
+    _ac: Optional[ActionClient] = None
```
Comment (nspielbau) [inline ros_bt_py/ros_bt_py/ros_nodes/action.py:501]: I think we can remove the `Client` import on l 44 now

File: ros_bt_py/ros_bt_py/ros_nodes/action.py [hunk 2]
```diff
@@ -498,7 +498,10 @@ class Action(Leaf):
...
+    _feedback = None
+    _internal_state = ActionStates.IDLE
+    passthrough: bool = False
```
Comment (nspielbau) [inline ros_bt_py/ros_bt_py/ros_nodes/action.py:504]: I am not quite sure what this does

File: ros_bt_py/ros_bt_py/tree_manager.py [hunk 3]
```diff
@@ -415,10 +413,19 @@ def __init__(
...
             return self.tree_msg.state
 
+    @typechecked
+    def set_state(self, new_state: str) -> None:
+        with self._state_lock:
+            self.tree_msg.state = new_state
+            self.ros_node.get_logger().fatal(
+                f"Updating tree state to {self.tree_msg.state}"
+            )
+
```
Comment (nspielbau) [inline ros_bt_py/ros_bt_py/tree_manager.py:428]: I like this fix, but I still found some lines with the old structure in the file. Is that on purpose?

File: ros_bt_py/ros_bt_py/tree_manager.py [hunk 4]
```diff
@@ -964,11 +969,7 @@ def control_execution(  # noqa: C901 # TODO: Remove this and simplfy the method.
             try:
                 root = self.find_root()
                 if root:
-                    with self._state_lock:
-                        self._setting_up = True
                     root.setup()
-                    with self._state_lock:
-                        self._setting_up = False
```
Comment (nspielbau) [inline ros_bt_py/ros_bt_py/tree_manager.py:971]: Are we just not locking on setup anymore?

File: ros_bt_py/ros_bt_py/tree_node.py [hunk 5]
```diff
@@ -430,12 +431,9 @@ def main(argv=None):
         executor = SingleThreadedExecutor()
         executor.add_node(tree_node)
 
-        try:
-            executor.spin()
-        finally:
-            executor.shutdown()
-            tree_node.destroy_node()
```
Comment (nspielbau) [inline ros_bt_py/ros_bt_py/tree_node.py:437]: Are these shutdowns what did the masking?

**Review by nspielbau — CHANGES_REQUESTED** [review]
Some clarification/cleanup, otherwise lgtm. 
Please also try to update the changelog so we don't have to do that afterwards. Maybe we should merge #103 before that tho :P

File: ros_bt_py/ros_bt_py/tree_manager.py (hunk 4 above)
Comment (Oberacda) [reply ros_bt_py/ros_bt_py/tree_manager.py:971]: As there are no read accesses to this variable it did not seemed to be used before.

File: ros_bt_py/ros_bt_py/tree_node.py (hunk 5 above)
Comment (Oberacda) [reply ros_bt_py/ros_bt_py/tree_node.py:437]: The finally clause did the masking. I removed it in favor of catching the KeyboardInterruption

File: ros_bt_py/ros_bt_py/tree_manager.py (hunk 3 above)
Comment (Oberacda) [reply ros_bt_py/ros_bt_py/tree_manager.py]: No I changed these occurences.

File: ros_bt_py/ros_bt_py/ros_nodes/action.py (hunk 2 above)
Comment (Oberacda) [reply ros_bt_py/ros_bt_py/ros_nodes/action.py]: Not read so I removed it. Everything still works, so it cannot have been that important.

Comment (nspielbau) [reply ros_bt_py/ros_bt_py/ros_nodes/action.py]: Famous last words :D

**Review by nspielbau — APPROVED** [review]
Tested shortly, still works -> LGTM

**Pull request merged**